go-controller/pkg/ovn/subnetallocator/allocator.go

```diff
@@ -55,35 +55,57 @@ func (sna *SubnetAllocator) MarkAllocatedNetwork(subnet *net.IPNet) error {
 	return fmt.Errorf("network %s does not belong to any known range", subnet.String())
 }
 
-func maybeAllocateOneNetwork(ranges []*subnetAllocatorRange, networks []*net.IPNet) ([]*net.IPNet, error) {
```

Comment:
IIUIC the slices are passed as reference, so this function was returning the network but also modifying the veriable.
Adding a method per IPFamily simplifies the logic later for `AddNode()`